docs/sources/reference/commandline/cli.md

```diff
@@ -637,13 +637,36 @@ an [*ADD*](/reference/builder/#add) instruction to reference a file in the
 context.
 
 The `URL` parameter can specify the location of a Git repository;
-the repository acts as the build context.  The system recursively clones the repository
+the repository acts as the build context. The system recursively clones the repository
 and its submodules using a `git clone --depth 1 --recursive` command.
 This command runs in a temporary directory on your local host.
 After the command succeeds, the directory is sent to the Docker daemon as the context.
 Local clones give you the ability to access private repositories using
 local user credentials, VPN's, and so forth.
 
+Git urls accept context configuration in their fragment section divided by a colon `:`.
+The first part represents the reference that git will check out, this can be either
+a branch, a tag or a commit sha. The second part represents a subdirectory
+inside the main root repository that will be used as a root context.
+
+For example, as context, you need to run this command to use a directory called `docker` in the branch `container`:
```

Comment:
(I've missed @moxiegirl's comments, so this may be for a reason); I'd leave out the "as context, you need to", and simplify to:

> For example, run this command to use a directory called `docker` in the branch `container`: